### PR TITLE
Allow scanning for metadata with occ scan:file --metadata

### PR DESCRIPTION
**Pull request by CarlSchwan**
This is an option since the scanning will take longer with the metadata scanning

Currently, the getNodeForPath function also needs to be fixed since the path manipulation with array_shift shouldn't be required, but I have no idea what to do instead. This was copied from the hookconnector class

### CODE REVIEW

File: apps/files/lib/Command/Scan.php [hunk 1]
```diff
@@ -312,4 +334,32 @@ protected function reconnectToDatabase(OutputInterface $output): Connection {
 		}
 		return $connection;
 	}
+
+	private function getNodeForPath($path) {
+		$pathParts = explode('/', $path);
+		// FIXME ugly hack to get it working for local file
+		array_shift($pathParts);
+		array_shift($pathParts);
+		array_shift($pathParts);
```
Comment (PVince81) [inline apps/files/lib/Command/Scan.php:343]: I think you can add an argument to `explode` to only do so for the first three or something and then keep the rest

did you check if using Filesystem::resolvePath would work here ?

Comment (PVince81) [inline apps/files/lib/Command/Scan.php:343]: I think you can add an argument to `explode` to only do so for the first three or something and then keep the rest

did you check if using Filesystem::resolvePath would work here ?

Comment (CarlSchwan) [reply apps/files/lib/Command/Scan.php:343]: The issue is that these array_shift shouldn't even exists :( This should work:

```php
 		$info = Filesystem::getFileInfo($path);
```

**Review by icewind1991 — CHANGES_REQUESTED** [review]

**Comment by matiasdelellis** [comment]
Just testing NC25...
Well, with the new view in the photos app, it should be integrated for that...

**Pull request closed without merging**